zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormSpanStore.scala

```diff
@@ -197,30 +197,44 @@ class AnormSpanStore(val db: DB,
       |WHERE t2.endpoint_service_name = {service_name}
     """.stripMargin
 
+  val allTracesHeader =
+    """SELECT t1.trace_id, start_ts
+      |FROM zipkin_spans t1
+      |JOIN zipkin_annotations t2 ON t1.trace_id = t2.trace_id AND t1.id = t2.span_id
+    """.stripMargin
+
   val sliceQueryFooter =
     """AND start_ts BETWEEN {start_ts} AND {end_ts}
       |GROUP BY t1.trace_id
       |ORDER BY start_ts DESC
       |LIMIT {limit}
     """.stripMargin
 
-  override def getTraceIdsByName(serviceName: String, spanName: Option[String],
+  override def getTraceIdsByName(serviceName: Option[String], spanName: Option[String],
     endTs: Long, lookback: Long, limit: Int): Future[Seq[IndexedTraceId]] = db.inNewThreadWithRecoverableRetry {
-
     if (endTs <= 0 || limit <= 0) {
       Seq.empty
     }
     else {
       implicit val (conn, borrowTime) = borrowConn()
       try {
-        val result: List[(Long, Long)] = SQL(sliceQueryHeader +
-          "AND (name = {name} OR {name} = '')" + sliceQueryFooter)
-          .on("service_name" -> serviceName)
-          .on("name" -> spanName.getOrElse(""))
-          .on("start_ts" -> (endTs - lookback) * 1000)
-          .on("end_ts" -> endTs * 1000)
-          .on("limit" -> limit)
-          .as((long("trace_id") ~ long("start_ts") map flatten) *)
+        val result:List[(Long, Long)] = serviceName match {
```

Comment:
It's conventional to have a space before the type declaration here:

    val result: List[(Long, Long)] = ...